shotover-proxy/tests/runner/observability_int_tests.rs

```diff
@@ -1,17 +1,75 @@
+        r#"shotover_transform_total{transform="QueryCounter"}"#,
+        r#"shotover_transform_total{transform="Null"}"#,
+        r#"# TYPE shotover_chain_failures counter"#,
+        r#"shotover_chain_failures{chain="redis_chain"}"#,
+        r#"# TYPE shotover_transform_failures counter"#,
+        r#"shotover_transform_failures{transform="Null"}"#,
+        r#"shotover_transform_failures{transform="QueryCounter"}"#,
+        r#"# TYPE shotover_available_connections gauge"#,
+        r#"shotover_available_connections{source="Redis Source"}"#,
+        r#"# TYPE shotover_transform_latency summary"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.5"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.9"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.95"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.99"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="0.999"}"#,
+        r#"shotover_transform_latency{transform="QueryCounter",quantile="1"}"#,
+        r#"shotover_transform_latency_sum{transform="QueryCounter"}"#,
+        r#"shotover_transform_latency_count{transform="QueryCounter"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.5"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.9"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.95"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.99"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="0.999"}"#,
+        r#"shotover_transform_latency{transform="Null",quantile="1"}"#,
+        r#"shotover_transform_latency_sum{transform="Null"}"#,
+        r#"shotover_transform_latency_count{transform="Null"}"#,
+        r#"# TYPE shotover_chain_latency summary"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.5"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.9"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.95"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.99"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="0.999"}"#,
+        r#"shotover_chain_latency{chain="redis_chain",quantile="1"}"#,
+        r#"shotover_chain_latency_sum{chain="redis_chain"}"#,
+        r#"shotover_chain_latency_count{chain="redis_chain"}"#,
+    ];
+    // check we get the metrics on startup
+
+    let mut body = http_request_metrics().await;
+
+    println!("{}", body);
+
+    for s in expected {
+        assert!(body.contains(s));
+    }
+
+    // Check we still get the metrics after sending a couple requests
+
     redis::cmd("SET")
```

Comment:
We can't register the `"shotover_chain_latency"` histogram metric because there will be a different label for every client that connects. I can:

- register it without the client_details label
- register it with an empty client_details, but that means that there will be a useless group of metrics with `client_details="" ` remaining in the metrics.
- not register it at all 

I think option 1 is the best.

Reply:
yes, lets remove client_details, as long thats there preregistering shotover_chain_latency just adds noise.

Reply:
To be clear, you mean just remove it from the registration? We can't remove it entirely.

Reply:
Yeah, I assume we need to remove it for the register to be useful.

> We can't remove it entirely.

why? is it important?

If it is needed then lets just not register it

Reply:
ok lets remove the register then